### ISSUE TITLE
    rx.exceptions.OnErrorNotImplementedException error on Android

### ISSUE THREAD

**Issue by SudoPlz** [Instabug/instabug-reactnative#74]
rx.exceptions.OnErrorNotImplementedException
        InternalObservableUtils.java:386 rx.internal.util.InternalObservableUtils$ErrorNotImplementedAction.call
        InternalObservableUtils.java:383 rx.internal.util.InternalObservableUtils$ErrorNotImplementedAction.call
        ActionSubscriber.java:44 rx.internal.util.ActionSubscriber.onError
        SafeSubscriber.java:152 rx.observers.SafeSubscriber._onError
        SafeSubscriber.java:115 rx.observers.SafeSubscriber.onError
        Exceptions.java:205 rx.exceptions.Exceptions.throwOrReport
        SafeSubscriber.java:139 rx.observers.SafeSubscriber.onNext
        PublishSubject.java:305 rx.subjects.PublishSubject$PublishSubjectProducer.onNext
        PublishSubject.java:220 rx.subjects.PublishSubject$PublishSubjectState.onNext
        PublishSubject.java:73 rx.subjects.PublishSubject.onNext
        EventBus.java:24 com.instabug.library.core.eventbus.a.a
        SDKCoreEventPublisher.java:9 com.instabug.library.core.eventbus.coreeventbus.b.a
        SessionManager.java:233 com.instabug.library.l.a
        SessionManager.java:202 com.instabug.library.l.k
        SessionManager.java:138 com.instabug.library.l.c
        SessionManager.java:269 com.instabug.library.l.a
        CurrentActivityLifecycleChanged.java:25 com.instabug.library.tracking.b.onReceive
        LocalBroadcastManager.java:297 android.support.v4.content.LocalBroadcastManager.executePendingBroadcasts
        LocalBroadcastManager.java:46 android.support.v4.content.LocalBroadcastManager.access$000
        LocalBroadcastManager.java:116 android.support.v4.content.LocalBroadcastManager$1.handleMessage
        Handler.java:102 android.os.Handler.dispatchMessage
        Looper.java:154 android.os.Looper.loop
        ActivityThread.java:6688 android.app.ActivityThread.main
        Method.java:-2 java.lang.reflect.Method.invoke
        ZygoteInit.java:1468 com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run
        ZygoteInit.java:1358 com.android.internal.os.ZygoteInit.main

    Caused by: java.util.ConcurrentModificationException
        HashMap.java:851 java.util.HashMap$HashIterator.nextEntry
        HashMap.java:891 java.util.HashMap$EntryIterator.next
        HashMap.java:890 java.util.HashMap$EntryIterator.next
        InstabugUserEventLogger.java:185 com.instabug.library.logging.InstabugUserEventLogger$1.a
        InstabugUserEventLogger.java:180 com.instabug.library.logging.InstabugUserEventLogger$1.call
        ActionSubscriber.java:39 rx.internal.util.ActionSubscriber.onNext
        SafeSubscriber.java:134 rx.observers.SafeSubscriber.onNext
        PublishSubject.java:305 rx.subjects.PublishSubject$PublishSubjectProducer.onNext
        PublishSubject.java:220 rx.subjects.PublishSubject$PublishSubjectState.onNext
        PublishSubject.java:73 rx.subjects.PublishSubject.onNext
        EventBus.java:24 com.instabug.library.core.eventbus.a.a
        SDKCoreEventPublisher.java:9 com.instabug.library.core.eventbus.coreeventbus.b.a
        SessionManager.java:233 com.instabug.library.l.a
        SessionManager.java:202 com.instabug.library.l.k
        SessionManager.java:138 com.instabug.library.l.c
        SessionManager.java:269 com.instabug.library.l.a
        CurrentActivityLifecycleChanged.java:25 com.instabug.library.tracking.b.onReceive
        LocalBroadcastManager.java:297 android.support.v4.content.LocalBroadcastManager.executePendingBroadcasts
        LocalBroadcastManager.java:46 android.support.v4.content.LocalBroadcastManager.access$000
        LocalBroadcastManager.java:116 android.support.v4.content.LocalBroadcastManager$1.handleMessage
        Handler.java:102 android.os.Handler.dispatchMessage
        Looper.java:154 android.os.Looper.loop
        ActivityThread.java:6688 android.app.ActivityThread.main
        Method.java:-2 java.lang.reflect.Method.invoke
        ZygoteInit.java:1468 com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run
        ZygoteInit.java:1358 com.android.internal.os.ZygoteInit.main

### SDK version
instabug-react-native: 2.0.3

**Comment by reemalashry1** [comment]
Hey @SudoPlz this issue seems to be caused by the native Android SDK, our Android team will be working on a fix, meanwhile, would you please tell me if this exception occurs consistently?

**Comment by SudoPlz** [comment]
I'm not entirely sure as our QA team is not testing Android yet.

**Comment by reemalashry1** [comment]
@SudoPlz No problem, we will be checking it and I'll get back to you 👍

**Comment by reemalashry1** [comment]
Hey @SudoPlz 
We have fixed the issue and made **4.2.11.1 Snapshot**.
Please add this to your app-level` build.gradle `file to grab the snapshot: 

```
dependencies { 
  compile 'com.instabug.library:instabug:4.2.11.1-SNAPSHOT' 
} 
repositories { 
    maven { 
        url "https://oss.sonatype.org/content/repositories/snapshots" 
    } 
} 

```

**Comment by SudoPlz** [comment]
Perfect, thanks!
